### ISSUE TITLE
JOSS Review

### ISSUE THREAD

**Issue by philippemiron** [AlignSAR/alignSAR#2]
Hi, I'm reviewing the repo for the JOSS publication (https://github.com/openjournals/joss-reviews/issues/7532#issuecomment-2549086865), and I have some comments. I summarized them into some itemized issues.

- [x] the package `AlignSAR` as a lot of scripts and examples, but the organization of the repo is hard to follow. Would it be possible to follow a more standard structure such has:
```
alignSAR/
|─── src/
│   └── alignsar/
│        └── __init__.py
│        └── core/
│        └── utils/
│        └── mlprocessing/
|─── paper/
│   └── flowchart.png (that is in the paper)
│   └── user_guide/
|─── examples/
│    └── data/
│    └── tutorials/
│    └── notebooks/
|─── dependencies/
│    └── doris_patched2023/
│    └── stac/
|─── tests/
|─── .env
|─── .gitignore
|─── Dockerfile
|─── pyproject.toml (or a requirements.txt)
|─── README.md
|─── LICENSE
|─── zenodo.json
```
- [x] include some tests to validate the scripts and utilities
- [x] is it possible to add citation of Doris and Snap in the paper
- [x] add `__pycache__` to `.gitignore` and remove from the repo
- [x] usually the `Dockerfile` is with this orthograph. This breaks the tutorial because when you execute `docker build -t alignsar .` it is looking for a `Dockerfile`.
- [x] environment variable in `bashrc_alignsar` can be set in a `.env` file and imported in the Dockerfile
- [x] Unable to build the image
```
git clone git@github.com:AlignSAR/alignSAR.git
cd alignSAR

~/repos/alignSAR on  main 
✗ docker build -f DockerFile -t alignsar .
[+] Building 0.0s (1/1) FINISHED                                                      docker:desktop-linux
 => [internal] load build definition from DockerFile                                                  0.0s
 => => transferring dockerfile: 4.38kB                                                                0.0s
DockerFile:24
--------------------
  23 |     pip install git+https://github.com/comet-licsar/licsar_extra.git
  24 | >>>   LiCSARpath="/opt/licsar_proc" && \
  25 | >>>   echo "export LiCSARpath=/opt/licsar_proc" >> ~/.bashrc && \
  26 | >>>   echo "export PATH=\$PATH:\$LiCSARpath/bin:\$LiCSARpath/bin/orig:\$LiCSARpath/bin/scripts:\$LiCSARpath/python" >> ~/.bashrc && \
  27 | >>>   echo "export PYTHONPATH=\$PYTHONPATH:\$LiCSARpath/python:\$LiCSARpath/python/LiCSAR_lib:\$LiCSARpath/python/LiCSAR_db" >> ~/.bashrc && \
  28 | >>>   echo "source activate py39" >> ~/.bashrc
  29 |     
--------------------
ERROR: failed to solve: dockerfile parse error on line 24: unknown instruction: LiCSARpath="/opt/licsar_proc"
```

Once this is fixed, I can test some of the utilities and finish the review. 

Cheers,

**Comment by LC-SAR** [comment]
thank you so much @philippemiron for your valuable suggestions. 
1. We have restructured the organization of the AlignSAR repo (https://github.com/AlignSAR/alignSAR) based on your suggestion. 
2. We have provided the test (to validate the scripts and utilities) in the folders 'examples' (https://github.com/AlignSAR/alignSAR/tree/main/examples) and 'tests' (https://github.com/AlignSAR/alignSAR/tree/main/tests), along with brief instruction (see the files readme / README under a selected folder). We have provided all the information on AlignSAR installation, methods, data used and step-by-step procedure for the use case/example in the tutorial ‘AlignSAR_tutorial.pdf’ (https://github.com/AlignSAR/alignSAR/tree/main/tutorial).
3. Line 31-39 (https://github.com/AlignSAR/alignSAR/blob/main/paper/paper.md), we added the references for doris, snap, gamma, asf services - on demand processing and LiCSAR 
4. We did add __pycache__ to .gitignore and remove it
5. We provided dockerfile installation in the ‘user_guide’ folder (https://github.com/AlignSAR/alignSAR/tree/main/paper/user_guide) and ‘README’ file (https://github.com/AlignSAR/alignSAR/blob/main/README.md), for the Dockerfile (https://github.com/AlignSAR/alignSAR/blob/main/Dockerfile). For instance, to build the dockerfile, run 'docker build -t alignsar .' and to test it, run 'docker run -it -v /your_local_path:/path_you_want_to_set_in_docker alignsar'. This dockerfile, as an example and demonstration, can automatically install Doris and relevant software. 
6. This bashrc_alignsar is associated with the docker file 'DockerFile_LiCSAR_SNAP' (https://github.com/AlignSAR/alignSAR/blob/main/DockerFile_LiCSAR_SNAP). This dockerfile can install LiCSAR and SNAP software. The methods and workflow description can be found in the tutorial ‘AlignSAR_tutorial.pdf’. In case of not building and running the Dockerfile(s), one can also manually install Doris, LiCSAR and SNAP and customize preferred processing environment. 
7.  We double-checked the two dockerfiles. For the DockerFile you ran, Line 23 (https://github.com/comet-licsar/licsar_extra.git) does not work, and we are working on it. As licsar_extra.git is not used for the AlignSAR version 1.1 (submitted for JOSS). Please test https://github.com/AlignSAR/alignSAR/blob/main/Dockerfile, by following the instruction we provided.

Thanks again and looking foward to your further suggestions, with regards,

Ling

**Comment by philippemiron** [comment]
Thanks for those updates. I have a few more suggestions to be able to close the last points.

- [ ] Installation instructions. Initially, building the Docker image failed, even specifying the platform, `docker build—-platform=linux/amd64 -t alignsar`. I get the following error:
```
206.4   Could not find a version that satisfies the requirement fiona (from versions: )
206.4 No matching distribution found for fiona
------
Dockerfile:12
--------------------
  11 |     # Build Python source for doirs
  12 | >>> RUN apt-get install -y gdal-bin python-gdal libgdal-dev gawk make tcsh csh wget git tar zip vim build-essential python-pip \
  13 | >>>   && pip install fiona shapely scipy html HTMLParser requests fastkml lxml -U pyopenssl matplotlib pyproj
  14 |     
```
- [ ] How is the user supposed to install Doris-5 installation which seems to be required to run the Docker image?
- [ ] Example usage and Automated tests. It would be ideal to include an `pyproject.toml`, or a conda `environment.yml in the example folders to be able to install quickly the required packages to run the example. Currently, this is left to the user to figure out. This also prevent me from validating the tests.

**Comment by LC-SAR** [comment]
Hi @philippemiron, thanks for your follow-up comments. Please see the response below:

1. Installation instructions: We have tested our dockerfile with three different operating systems/docker versions: macOS (docker version: version 4.37.2), ubuntu 20.04.6 LTS (docker version: 24.0.5) and Ubuntu 20.04.1 (docker version: 24.0.7), we didn’t encounter any issues when building/running dockerfile. 
For instance, with macOS (Ventura 13.7.2), docker-desktop app (version 4.37.2(179585)), here are the test log, see the screenshot below. 
![Screenshot 2025-01-13 at 14 37 36](https://github.com/user-attachments/assets/bf0ec211-33b8-4032-8aed-3f50317e5190)
We didn’t figure out why you had this error and how to solve it.
 2. How is the user supposed to install Doris-5 installation: We have addressed that ‘_Note that AlignSAR provides SAR data preprocessing workflow using 1) Doris, and 2) LiCSAR and SNAP. ‘Dockerfile’ can install Doris-5 and its relevant third-party software tools, while DockerFile_LiCSAR_SNAP is another dockerfile we offered, which can install LiCSAR and SNAP software. **The methods and workflow description can be found in the tutorial ‘AlignSAR_tutorial.pdf’. In case of not building and running the Dockerfile(s), one can also manually install Doris, LiCSAR and SNAP and customize preferred processing environment.**_’ in https://github.com/AlignSAR/alignSAR/blob/main/README.md. And in the tutorial https://github.com/AlignSAR/alignSAR/blob/main/tutorial/AlignSAR_tutorial.pdf, Appendix A (page 30) describes how to manually install Doris. 
 3. Example usage and Automated tests: We created pyproject.toml file, see https://github.com/AlignSAR/alignSAR/blob/main/pyproject.toml. Note that Doris-5 is built upon Python version-2.7.17, which is used to generate SAR signatures (such as interferograms and amplitude maps). The procedure is illustrated in the tutorial https://github.com/AlignSAR/alignSAR/blob/main/tutorial/AlignSAR_tutorial.pdf, Chapter 3 ‘Procedure description’. In https://github.com/AlignSAR/alignSAR/blob/main/examples/data/data_links.txt, for instance, we shared the samples of the resultant SAR benchmark datasets (using Doris) for the use case in the Netherlands (item 5). You can test them with jupyter_notebook_demo (https://github.com/AlignSAR/alignSAR/tree/main/examples/jupyter_notebook_demo). The rest of the tools we created are built upon Python version 3.10, therefore in project.toml, we list two Python language versions: 2.7.17 and 3.10.
